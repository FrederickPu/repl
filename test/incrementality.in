{"cmd": "example : True := by\n  dbg_trace \"!\"\n  sorry"}

{"cmd": "example : True := by\n  dbg_trace \"!\"\n  sorry"}

{"cmd": "example : True := by\n  dbg_trace \"!\"\n  sorry", "incr": 1}

{"cmd": "example : True := by\n  dbg_trace \"!\"\n  sorry", "incr": 37}

