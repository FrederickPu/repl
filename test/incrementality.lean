import Lean

open Lean

example : True := by
  dbg_trace "!"
  sorry
