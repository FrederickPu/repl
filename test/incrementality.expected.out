!
{"results":
 [{"source": "example : True := by\n  dbg_trace \"!\"\n  sorry",
   "sorries":
   [{"proofState": 0,
     "pos": {"line": 3, "column": 2},
     "goals": "⊢ True",
     "endPos": {"line": 3, "column": 7}}],
   "messages":
   [{"severity": "warning",
     "pos": {"line": 1, "column": 0},
     "endPos": {"line": 1, "column": 7},
     "data": "declaration uses 'sorry'"}],
   "env": 0}]}

{"results":
 [{"source": "example : True := by\n  dbg_trace \"!\"\n  sorry",
   "sorries":
   [{"proofState": 1,
     "pos": {"line": 3, "column": 2},
     "goals": "⊢ True",
     "endPos": {"line": 3, "column": 7}}],
   "messages":
   [{"severity": "warning",
     "pos": {"line": 1, "column": 0},
     "endPos": {"line": 1, "column": 7},
     "data": "declaration uses 'sorry'"}],
   "env": 1}]}

{"results":
 [{"source": "example : True := by\n  dbg_trace \"!\"\n  sorry",
   "sorries":
   [{"proofState": 2,
     "pos": {"line": 3, "column": 2},
     "goals": "⊢ True",
     "endPos": {"line": 3, "column": 7}}],
   "messages":
   [{"severity": "warning",
     "pos": {"line": 1, "column": 0},
     "endPos": {"line": 1, "column": 7},
     "data": "declaration uses 'sorry'"}],
   "env": 2}]}

{"message": "Unknown incremental state 37"}

